PFW-1189 Pre-release tasks

3d-gussner opened the pull request:
- [X] Comment messages `c= r=` for translations
- [X] RepRap documentation
   - [X] [M123](https://www.reprap.org/wiki/G-code#M123:_Tachometer_value_.28RepRap_.26_Prusa.29)
   - [X] [M155](https://www.reprap.org/wiki/G-code#M155:_Automatically_send_temperatures)
   - [X] [M20 L T](https://www.reprap.org/wiki/G-code#M20:_List_SD_card) 
   - [X] [M73 C D](https://www.reprap.org/wiki/G-code#M73:_Set.2FGet_build_percentage)
   - [X] Update capabilities

- [x] Update translations
   - [x] Czech
   - [X] German
   - [x] Spanish
   - [x] French **Thanks to @carlin57 & @awenelo** 
   - [X] Italian **Thanks to @wavexx** 
   - [x] Polish
   - [X] Community made
      - [X] Dutch

- [x] Update po files
- [X] _Review & update `translation.md`_ Move to FW3.11.0
- [X] _Review & update `Add_new_language.md`_ Move to FW3.11.0
- [X] _Review & update `readme.md_` Move to FW3.11.0
- [X] Review & cleanup `//@todo` in source code related to gcode

3d-gussner committed:
Update MSG_SELFTEST_PLEASECHECK c=20

3d-gussner committed:
Update MSG_SELFTEST_ENDSTOPS c=20

3d-gussner committed:
Update MSG_SELFTEST_ENDSTOP c=16

3d-gussner committed:
Update MSG_LOOSE_PULLEY c=20

3d-gussner committed:
Update MSG_BED_DONE c=20

3d-gussner committed:
Update MSG_BED_HEATING c=20

3d-gussner committed:
Update #MSG_BED_CORRECTION_MENU c=18

3d-gussner committed:
Update #MSG_BED c=13

3d-gussner committed:
Update #MSG_CALIBRATE_BED c=18

3d-gussner committed:
Update #MSG_HOMEYZ c=18

3d-gussner committed:
Update #MSG_HOMEYZ_DONE c=20

3d-gussner committed:
Update #MSG_MENU_CALIBRATION c=18

3d-gussner committed:
Update #MSG_SD_REMOVED c=20

3d-gussner committed:
Define MSG_CHECKING_FILE c=17

3d-gussner committed:
Update #MSG_NOT_COLOR c=19

3d-gussner committed:
Update #MSG_COOLDOWN c=18

3d-gussner committed:
Define #MSG_COPY_SEL_LANG c=20 r=3

3d-gussner committed:
Define #MSG_CALIBRATING_HOME c=20

3d-gussner committed:
Define #MSG_CHOOSE_FIL_1ST_LAYERCAL c=20 r=7

3d-gussner committed:
Define #MSG_CRASH_RESUME c=20 r=3

3d-gussner committed:
Update #MSG_DISABLE_STEPPERS c=18

3d-gussner committed:
Define #MSG_MMU_CUTTING_FIL c=18

3d-gussner committed:
Update #MSG_ERROR c=10

3d-gussner committed:
Define #MSG_MMU_FAIL_STATS c=18

3d-gussner committed:
Define #MSG_FAIL_STATS c=18

3d-gussner committed:
Update #MSG_FSENSOR c=12

3d-gussner committed:
Define #MSG_MMU_FIX_ISSUE c=20 r=4

3d-gussner committed:
Update #MSG_FLOW c=15

3d-gussner committed:
Update #MSG_PRUSA3D_FORUM c=18

3d-gussner committed:
Update #MSG_PRUSA3D c=18

3d-gussner committed:
Update #MSG_PRUSA3D_HOWTO c=18

3d-gussner committed:
Update #MSG_HEATING c=20

3d-gussner committed:
Update #MSG_FILAMENTCHANGE c=18

3d-gussner committed:
Update #MSG_CHANGE_SUCCESS c=20

3d-gussner committed:
Update #MSG_WATCH c=18

3d-gussner committed:
Define #MSG_ADDITIONAL_SHEETS c=20 r=9

3d-gussner committed:
Define #MSG_LEFT c=10

3d-gussner committed:
Define #MSG_RIGHT c=10

3d-gussner committed:
Update MSG_LANGUAGE_SELECT c=18

3d-gussner committed:
Update #MSG_SELFTEST_OK c=20

3d-gussner committed:
Update #MSG_SELFTEST c=18

3d-gussner committed:
Update #MSG_BELTTEST c=18

3d-gussner committed:
Define #MSG_SEL_PREHEAT_TEMP c=20 r=6

3d-gussner committed:
Update #MSG_SETTINGS c=18

3d-gussner committed:
Update #MSG_SORT c=7

3d-gussner committed:
Update #MSG_SORT_ALPHA c=8 #MSG_SORT_TIME c=8

3d-gussner committed:
Update #MSG_RPI_PORT c=13

3d-gussner committed:
Update #MSG_SOUND c=9

3d-gussner committed:
Update #MSG_SOUND_BLIND c=7, #MSG_SOUND_LOUD c=7, MSG_SOUND_ONCE c=7

3d-gussner committed:
Update #MSG_MAIN c=18

3d-gussner committed:
Update #MSG_NONE c=8

3d-gussner committed:
Update #MSG_WARN c=8

3d-gussner committed:
Update #MSG_STRICT c=8

3d-gussner committed:
Update #MSG_MODEL c=8

3d-gussner committed:
Update #MSG_FIRMWARE c=8

3d-gussner committed:
Update #MSG_NOZZLE_DIAMETER c=10, MSG_GCODE c=8

3d-gussner committed:
Update #MSG_MMU_MODE c=8

3d-gussner committed:
Update #MSG_SD_CARD c=8

3d-gussner committed:
Update #MSG_TOSHIBA_FLASH_AIR_COMPATIBILITY c=8

3d-gussner committed:
Update #MSG_MESH c=12

3d-gussner committed:
Update #MSG_Z_PROBE_NR c=14

3d-gussner committed:
Update #MSG_MAGNETS_COMP c=13

3d-gussner committed:
Update #MSG_CARD_MENU c=18

3d-gussner committed:
Update MSG_BACK c=18

3d-gussner committed:
Update #MSG_NOZZLE c=12

3d-gussner committed:
Update MSG_REFRESH c=18

3d-gussner committed:
Update MSG_RESUMING_PRINT c=20

3d-gussner committed:
Update #MSG_SELFTEST_MOTOR c=18

3d-gussner committed:
Update #MSG_SELFTEST_WIRINGERROR c=17

3d-gussner committed:
Update #MSG_MODE c=9

3d-gussner committed:
Update #MSG_MODE c=6

3d-gussner committed:
Update #MSG_HIGH_POWER c=10

3d-gussner committed:
Update #MSG_AUTO_POWER c=10

3d-gussner committed:
Update #MSG_STEALTH c=7, #MSG_SILENT c=7, #MSG_NORMAL c=7

3d-gussner committed:
Update #MSG_STOP_PRINT c=18

3d-gussner committed:
Update #MSG_STOPPED c=20

3d-gussner committed:
Update #MSG_NA c=3, #MSG_OFF c=3, #MSG_ON c=3

3d-gussner committed:
Update #MSG_AUTO_DEPLETE c=13

3d-gussner committed:
Update #MSG_NO c=3

3d-gussner committed:
Update #MSG_BABYSTEP_Z c=18

3d-gussner committed:
Update MSG_LOADING_COLOR c=20

3d-gussner committed:
Update #MSG_LOAD_TO_NOZZLE c=18

3d-gussner committed:
Update #MSG_MEASURE_BED_REFERENCE_HEIGHT_LINE1 c=20 r=3

3d-gussner committed:
Define #MSG_MEASURED_SKEW c=13

3d-gussner committed:
Define MSG_MMU_LOAD_FAILED c=20

3d-gussner committed:
Define #MSG_MMU_CONNECTED c=18

3d-gussner committed:
Update #MSG_MOVE_AXIS c=18

3d-gussner committed:
Update MSG_MOVE_X/Y/Z c=18

3d-gussner committed:
Update #MSG_NO_MOVE c=20

3d-gussner committed:
Update #MSG_YES c=3

3d-gussner committed:
Define #MSG_Y_DIST_FROM_MIN c=20

3d-gussner committed:
Update #MSG_FIL_FAILD c=20 r=5

3d-gussner committed:
Define #MSG_GCODE_DIFF_CONTINUE c=20 r=4, #MSG_GCODE_DIFF_CANCELLED c=20 r=7

3d-gussner committed:
Define #MSG_GCODE_NEWER_FIRMWARE_CONTINUE c=20 r=5, #MSG_GCODE_NEWER_FIRMWARE_CANCELLED c=20 r=8

3d-gussner committed:
Define #MSG_PREHEATING_TO_LOAD/UNLOAD/CUT/EJECT c=20

3d-gussner committed:
Define #MSG_TO_LOAD_FIL c=20, #MSG_TO_UNLOAD_FIL c=20, #MSG_PRESS_KNOB c=20

3d-gussner committed:
Define #MSG_FS_VERIFIED c=20 r=3

3d-gussner committed:
Define #MSG_LIN_CORRECTION c=18

3d-gussner committed:
Define #MSG_INSERT_FIL c=20 r=6

3d-gussner committed:
Define #MSG_REMOVE_TEST_PRINT c=20 r=4

3d-gussner committed:
Define #MSG_PINDA c=6 and remove duplicate saved 8 bytes

3d-gussner committed:
Fix #MSG_CRASHDETECT c=13 in Polish lang file

3d-gussner committed:
Update MSG_NO_CARD c=18

3d-gussner committed:
Define #MSG_NOZZLE_FAN c=10, #MSG_PRINT_FAN c=10

3d-gussner committed:
Define #MSG_REMOVE_SHIPPING_HELPERS c=20 r=3

3d-gussner committed:
Update MSG_PREHEAT c=18

3d-gussner committed:
Update #MSG_TUNE c=18, #MSG_PREHEAT c=18

3d-gussner committed:
Define #MSG_RESUME_NOZZLE_TEMP c=20 r=4

3d-gussner committed:
Define #MSG_MMU_INSERT_FILAMENT_FIRST_TUBE c=20 r=6

3d-gussner committed:
Define #MSG_UNLOAD_FILAMENT_REPEAT c=20 r=4

3d-gussner committed:
Define #MSG_CHECK_IR_CONNECTION c=20 r=4

3d-gussner committed:
Define #MSG_SLIGHT_SKEW c=13, #MSG_SEVERE_SKEW c=13

3d-gussner committed:
Update #MSG_SOUND_LOUD c=7

3d-gussner committed:
Define #MSG_RUNOUTS c=7

3d-gussner committed:
Define #MSG_Z-LEVELING_ENFORCED c=20 r=4

3d-gussner committed:
Update #MSG_SPEED c=15

3d-gussner committed:
Update #MSG_STATISTICS c=18

3d-gussner committed:
Update #MSG_SUPPORT c=18

3d-gussner committed:
Define #MSG_SELECT_FILAMENT c=20

3d-gussner committed:
Define #MSG_SELECT_TEMP_MATCHES_MATERIAL c=20 r=4

3d-gussner committed:
Update MSG_TEMPERATURE c=18

3d-gussner committed:
Define and update #MSG_PRINT_TIME c=19, #MSG_FILAMENT_USED c=19, #MSG_TOTAL_FILAMENT c=19, #MSG_TOTAL_PRINT_TIME c=19

3d-gussner committed:
Delete "Unload" as it isn't used anymore

3d-gussner committed:
Define #MSG_UNKNOWN c=13

3d-gussner committed:
Update #MSG_BACK c=18

3d-gussner committed:
Define #MSG_CHECKS c=18

3d-gussner committed:
Define #MSG_FALSE_TRIGGERING c=20

3d-gussner committed:
Define #MSG_FINDA c=5 and update #MSG_PINDA c=5

3d-gussner committed:
Define #MSG_MK3S_FIRMWARE_ON_MK3 c=20 r=4, #MSG_MK3_FIRMWARE_ON_MK3S c=20 r=4

3d-gussner committed:
Define #MSG_MODE_CHANGE_IN_PROGRESS c=20 r=3

3d-gussner committed:
Define #MSG_NOZZLE_DIFFERS_CONTINUE c=20 r=5, #MSG_NOZZLE_DIFFERS_CANCELLED c=20 r=9

3d-gussner committed:
Define #MSG_RENAME c=18

3d-gussner committed:
Define #MSG_SELECT c=18

3d-gussner committed:
Update #MSG_INFO_SENSORS c=18

3d-gussner committed:
Fix Dutch #MSG_FINDA c=5

3d-gussner committed:
Remove r=1 and some minor fixes

3d-gussner committed:
Cleanup some comments ////c=

3d-gussner committed:
Define #MSG_PRINTER_IP c=18

3d-gussner committed:
Fix output "Y distance from min" by removing ":" so translations can use 20 chars

3d-gussner committed:
Updated RepRap Wiki and removed the @todos

3d-gussner committed:
Update RepRap url for `M123`

3d-gussner committed:
Merge remote-tracking branch 'upstream/MK3_3.10.0' into PFW-1189

3d-gussner committed:
Fix merge issues

3d-gussner commented:
We have found few translations that are too long. @awenelo @carlin57 as you helped last time I would like to ask you if you would mind to review these French translations?

```
[E]: Text "Corr. niveau plateau" is longer then definition on line 109 (rows diff=0)
[EN]:Text "Bed level correct" cols=18 rows=1

[E]: Text "Deviat.mesuree" is longer then definition on line 601 (rows diff=0)
[EN]:Text "Measured skew" cols=13 rows=1

[E]: Text "Echecs charg. MMU" is longer then definition on line 613 (rows diff=0)
[EN]:Text "MMU load fails" cols=15 rows=1

[E]: Text "Echecs alim. MMU" is longer then definition on line 641 (rows diff=0)
[EN]:Text "MMU power fails" cols=15 rows=1

[E]: Text "Haute puiss" is longer then definition on line 653 (rows diff=0)
[EN]:Text "High power" cols=10 rows=1

[E]: Text "Reprendre impression" is longer then definition on line 933 (rows diff=0)
[EN]:Text "Resume print" cols=18 rows=1

[E]: Text "Recherche du point de calibration du plateau chauffant" is longer then definition on line 965 (rows diff=1)
[EN]:Text "Searching bed calibration point" cols=20 rows=2

[E]: Text "Une fois" is longer then definition on line 1073 (rows diff=0)
[EN]:Text "Once" cols=7 rows=1

[E]: Text "Selectionnez le filament:" is longer then definition on line 1109 (rows diff=1)
[EN]:Text "Select filament:" cols=20 rows=1

End fr lang-check
```

3d-gussner committed:
Use textwarp to check the message length and rows
@DRracer: Added origial English text to the output, added cols and rows as well

3d-gussner committed:
Update "\04Refresh" message as two "abcd" "efgh" in one line cause issues in lang-build scripts

3d-gussner committed:
Readded `>Cancel` as the `>` has some meaning. Sadly the previously 10bytes more free space are gone again

3d-gussner commented:
If someone else form the community is willing to review too long translations, here a list of all
```
Start en lang-check
End en lang-check
count=364
offs=744
size=10203
lang=cz
Start cz lang-check
End cz lang-check
count=364
offs=744
size=9665
lang=de
Start de lang-check
End de lang-check
count=364
offs=744
size=11212
lang=es
Start es lang-check
[E]: Text "Calibrar posicion inicial" is longer then definition on line 141 (rows diff=1)
[EN]:Text "Calibrating home" cols=20 rows=1

[E]: Text "ERROR: El sensor de filamento no responde, por favor comprueba la conexion." is longer then definition on line 289 (rows diff=1)
[EN]:Text "ERROR: Filament sensor is not responding, please check connection." cols=20 rows=4

[E]: Text "Estadistica de fallos MMU" is longer then definition on line 313 (rows diff=1)
[EN]:Text "Fail stats MMU" cols=18 rows=1

[E]: Text "Estadistica de fallos" is longer then definition on line 321 (rows diff=1)
[EN]:Text "Fail stats" cols=18 rows=1

[E]: Text "Ventiladores frontal/izquierdo" is longer then definition on line 405 (rows diff=1)
[EN]:Text "Front/left fans" cols=20 rows=1

[E]: Text "Correccion de Linealidad" is longer then definition on line 537 (rows diff=1)
[EN]:Text "Lin. correction" cols=18 rows=1

[E]: Text "Cargar a la boquilla" is longer then definition on line 565 (rows diff=0)
[EN]:Text "Load to nozzle" cols=18 rows=1

[E]: Text "Desviacion med:" is longer then definition on line 601 (rows diff=0)
[EN]:Text "Measured skew" cols=13 rows=1

[E]: Text "Silencio" is longer then definition on line 633 (rows diff=0)
[EN]:Text "Silent" cols=7 rows=1

[E]: Text "Fallo de energia en MMU" is longer then definition on line 641 (rows diff=1)
[EN]:Text "MMU power fails" cols=15 rows=1

[E]: Text "Silencio" is longer then definition on line 645 (rows diff=0)
[EN]:Text "Stealth" cols=7 rows=1

[E]: Text "Fuerza auto" is longer then definition on line 649 (rows diff=0)
[EN]:Text "Auto power" cols=10 rows=1

[E]: Text "Distancia en Y desde el min" is longer then definition on line 1293 (rows diff=1)
[EN]:Text "Y distance from min" cols=20 rows=1

[E]: Text "Diam. nozzl" is longer then definition on line 1373 (rows diff=0)
[EN]:Text "Nozzle d." cols=10 rows=1

[E]: Text "Asistido" is longer then definition on line 1437 (rows diff=0)
[EN]:Text "Assist" cols=7 rows=1

End es lang-check
count=364
offs=744
size=10846
lang=fr
Start fr lang-check
[E]: Text "La detection de\x0acrash peut etre\x0aactive seulement en\x0amode Normal" is longer then definition on line 33 (rows diff=1)
[EN]:Text "Crash detection can\x0abe turned on only in\x0aNormal mode" cols=20 rows=4

[E]: Text "Corr. niveau plateau" is longer then definition on line 109 (rows diff=0)
[EN]:Text "Bed level correct" cols=18 rows=1

[E]: Text "Deviat.mesuree" is longer then definition on line 601 (rows diff=0)
[EN]:Text "Measured skew" cols=13 rows=1

[E]: Text "Echecs charg. MMU" is longer then definition on line 613 (rows diff=0)
[EN]:Text "MMU load fails" cols=15 rows=1

[E]: Text "Echecs alim. MMU" is longer then definition on line 641 (rows diff=0)
[EN]:Text "MMU power fails" cols=15 rows=1

[E]: Text "Haute puiss" is longer then definition on line 653 (rows diff=0)
[EN]:Text "High power" cols=10 rows=1

[E]: Text "Reprendre impression" is longer then definition on line 933 (rows diff=0)
[EN]:Text "Resume print" cols=18 rows=1

[E]: Text "Recherche du point de calibration du plateau chauffant" is longer then definition on line 965 (rows diff=1)
[EN]:Text "Searching bed calibration point" cols=20 rows=2

[E]: Text "Une fois" is longer then definition on line 1073 (rows diff=0)
[EN]:Text "Once" cols=7 rows=1

[E]: Text "Selectionnez le filament:" is longer then definition on line 1109 (rows diff=1)
[EN]:Text "Select filament:" cols=20 rows=1

End fr lang-check
count=364
offs=744
size=11433
lang=it
Start it lang-check
[E]: Text "Caricamento automatico attivo, premi la manopola e inserisci il filamento." is longer then definition on line 85 (rows diff=1)
[EN]:Text "Autoloading filament is active, just press the knob and insert filament..." cols=20 rows=4

[E]: Text "Calibrazione completa" is longer then definition on line 173 (rows diff=1)
[EN]:Text "Calibration done" cols=20 rows=1

[E]: Text "Ventola frontale/sinistra" is longer then definition on line 405 (rows diff=1)
[EN]:Text "Front/left fans" cols=20 rows=1

[E]: Text "Deviazione mis" is longer then definition on line 601 (rows diff=0)
[EN]:Text "Measured skew" cols=13 rows=1

[E]: Text "Caricamento MMU fallito" is longer then definition on line 609 (rows diff=1)
[EN]:Text "MMU load failed     " cols=20 rows=1

[E]: Text "Caricamenti MMU falliti" is longer then definition on line 613 (rows diff=1)
[EN]:Text "MMU load fails" cols=15 rows=1

[E]: Text "Silenzioso" is longer then definition on line 633 (rows diff=0)
[EN]:Text "Silent" cols=7 rows=1

[E]: Text "Silenziosa" is longer then definition on line 645 (rows diff=0)
[EN]:Text "Stealth" cols=7 rows=1

[E]: Text "Nuova versione firmware disponibile:" is longer then definition on line 701 (rows diff=1)
[EN]:Text "New firmware version available:" cols=20 rows=2

[E]: Text "Ventola estrusore" is longer then definition on line 729 (rows diff=0)
[EN]:Text "Nozzle FAN" cols=10 rows=1

[E]: Text "Mancanza corrente" is longer then definition on line 841 (rows diff=0)
[EN]:Text "Power failures" cols=15 rows=1

[E]: Text "Ventola di stampa" is longer then definition on line 881 (rows diff=0)
[EN]:Text "Print FAN" cols=10 rows=1

[E]: Text "Reset calibrazione XYZ." is longer then definition on line 925 (rows diff=1)
[EN]:Text "Reset XYZ calibr." cols=18 rows=1

[E]: Text "Ricerca dei punti di calibrazione del piano" is longer then definition on line 965 (rows diff=1)
[EN]:Text "Searching bed calibration point" cols=20 rows=2

[E]: Text "Seleziona il filamento:" is longer then definition on line 1109 (rows diff=1)
[EN]:Text "Select filament:" cols=20 rows=1

[E]: Text "Scarica filamento" is longer then definition on line 1173 (rows diff=0)
[EN]:Text "Unload filament" cols=16 rows=1

[E]: Text "Diam.Ugello" is longer then definition on line 1373 (rows diff=0)
[EN]:Text "Nozzle d." cols=10 rows=1

End it lang-check
count=364
offs=744
size=10966
lang=pl
Start pl lang-check
[E]: Text "Wykrywanie zderzen\x0amoze byc wlaczone\x0atylko w\x0atrybie Normalnym" is longer then definition on line 33 (rows diff=1)
[EN]:Text "Crash detection can\x0abe turned on only in\x0aNormal mode" cols=20 rows=4

[E]: Text "Przedni/lewy wentylator" is longer then definition on line 405 (rows diff=1)
[EN]:Text "Front/left fans" cols=20 rows=1

[E]: Text "Zmierzony skos" is longer then definition on line 601 (rows diff=0)
[EN]:Text "Measured skew" cols=13 rows=1

[E]: Text "Bledy ladow. MMU" is longer then definition on line 613 (rows diff=0)
[EN]:Text "MMU load fails" cols=15 rows=1

[E]: Text "Normalni" is longer then definition on line 629 (rows diff=0)
[EN]:Text "Normal" cols=7 rows=1

[E]: Text "Zaniki zasil. MMU" is longer then definition on line 641 (rows diff=0)
[EN]:Text "MMU power fails" cols=15 rows=1

[E]: Text "Wysoka wyd." is longer then definition on line 653 (rows diff=0)
[EN]:Text "High power" cols=10 rows=1

[E]: Text "Wstrzymanie wydruku" is longer then definition on line 733 (rows diff=0)
[EN]:Text "Pause print" cols=18 rows=1

[E]: Text "Prosze zaktualizowac." is longer then definition on line 829 (rows diff=1)
[EN]:Text "Please upgrade." cols=20 rows=1

[E]: Text "Zaniki zasilania" is longer then definition on line 841 (rows diff=0)
[EN]:Text "Power failures" cols=15 rows=1

[E]: Text "Szukam punktu kalibracyjnego na stole" is longer then definition on line 965 (rows diff=1)
[EN]:Text "Searching bed calibration point" cols=20 rows=2

[E]: Text "Sortowanie" is longer then definition on line 1029 (rows diff=0)
[EN]:Text "Sort" cols=7 rows=1

[E]: Text "Rozladowanie fil." is longer then definition on line 1173 (rows diff=0)
[EN]:Text "Unload filament" cols=16 rows=1

[E]: Text "Czekam na uzytkownika..." is longer then definition on line 1197 (rows diff=1)
[EN]:Text "Wait for user..." cols=20 rows=1

End pl lang-check
count=364
offs=744
size=10270
lang-build.sh finished with success
lang-community.sh started
 Source code path: OK
 Found: 
  NL

lang-build.sh started
lang=nl
Start nl lang-check
End nl lang-check
count=364
offs=744
size=10315
lang-build.sh finished with success
```

3d-gussner committed:
Fix Typo in MSG_FIL_FAILED

3d-gussner committed:
Added ` EXTENDED_M20` to RepRap Gcode wiki `M115` see https://reprap.org/mediawiki/index.php?title=G-code&type=revision&diff=189034&oldid=189033

wavexx committed:
Allow to customize OUTDIR in config.sh

Useful for custom build trees.

wavexx committed:
Be more strict when checking translation lenght

Strip quotes immediately so that we can measure strings easily.
For single-line strings, check the translation lenght correctly.

wavexx committed:
Do not output colors when redirecting output

wavexx committed:
Remove warnings caused by '\x00'

Manually unescape '\x00' for the time being, removing spurious warnings.

Commit:
Fix length of some CZ messages

3d-gussner committed:
@leptun strip 1st char hack for duplicate `Cancel`

3d-gussner committed:
Merge branch 'PFW-1189' of https://github.com/3d-gussner/Prusa-Firmware into PFW-1189

3d-gussner committed:
Fix MSG_UNLOAD_FILAMENT length as in MMU a number is added

3d-gussner committed:
Fix Dutch +  German MSG_UNLOAD_FILAMENT too long translations

3d-gussner committed:
Fix German MSG_LOAD_TO_NOZZLE

3d-gussner committed:
Fix MSG_NO being too short, no need to be

3d-gussner committed:
Fix MSG_SELFTEST_WIRINGERROR to c=18

3d-gussner committed:
Fix German too long translation MSG_MMU_MODE

3d-gussner committed:
Fix typo thanks to @wavexx

wavexx committed:
lang-check: print a warning if cols!=20 on multiline messages

wavexx committed:
lang-check: simplify wrapping, wrap according to cols

wavexx committed:
lang-check: pretty-print errors to aid in translation

carlin57 commented:
> We have found few translations that are too long. @awenelo @carlin57 as you helped last time I would like to ask you if you would mind to review these French translations?
> 
> ```
> [E]: Text "Corr. niveau plateau" is longer then definition on line 109 (rows diff=0)
> [EN]:Text "Bed level correct" cols=18 rows=1
>
 
> [E]: Text "Deviat.mesuree" is longer then definition on line 601 (rows diff=0)
> [EN]:Text "Measured skew" cols=13 rows=1
> 
> [E]: Text "Echecs charg. MMU" is longer then definition on line 613 (rows diff=0)
> [EN]:Text "MMU load fails" cols=15 rows=1
>  "Def. charg. MMU" 

> [E]: Text "Echecs alim. MMU" is longer then definition on line 641 (rows diff=0)
> [EN]:Text "MMU power fails" cols=15 rows=1
> "Def. alim. MMU"

> [E]: Text "Haute puiss" is longer then definition on line 653 (rows diff=0)
> [EN]:Text "High power" cols=10 rows=1
> 
> [E]: Text "Reprendre impression" is longer then definition on line 933 (rows diff=0)
> [EN]:Text "Resume print" cols=18 rows=1
> "Reprise impression"

> [E]: Text "Recherche du point de calibration du plateau chauffant" is longer then definition on line 965 (rows diff=1)
> [EN]:Text "Searching bed calibration point" cols=20 rows=2
> 

> [E]: Text "Une fois" is longer then definition on line 1073 (rows diff=0)
> [EN]:Text "Once" cols=7 rows=1
> 

> [E]: Text "Selectionnez le filament:" is longer then definition on line 1109 (rows diff=1)
> [EN]:Text "Select filament:" cols=20 rows=1
> "Choisissez filament"

> End fr lang-check
> ```

wavexx committed:
Update Italian translations

wavexx committed:
lang-check: highlight truncated strings

wavexx committed:
lang-check: handle backslash sequences

3d-gussner committed:
Fix Dutch FINDA translation

awenelo reviewed (changes requested):
See below.

awenelo commented on lang/lang_en_fr.txt:
```diff
@@ -94,36 +86,36 @@
-#MSG_BED_CORRECTION_MENU
+#MSG_BED_CORRECTION_MENU c=18
 "Bed level correct"
 "Corr. niveau plateau"
```
```suggestion
"Chg niveau plateau"
```
(18)
Changing "Correct" to "Change"(`Changer`), which then allows `Changer` to be shortened to `Chg`

carlin57 replied:
"Reglage plateau"

3d-gussner replied:
Changed it to "Reglage plateau"

awenelo commented on lang/lang_en_fr.txt:
```diff
@@ -606,15 +598,15 @@
-#
+#MSG_MEASURED_SKEW c=13
 "Measured skew"
 "Deviat.mesuree"
```
```suggestion
"Var. mesuree"
```
(12)
`Deviation` can be replaced by `Variation`, which can then be shortened to `Var`.

carlin57 replied:
or just "deviation"

awenelo replied:
The shortest shortening of `Deviation` that can be done and still makes sense to me is `Deviat.`, which is why I didn't want to shorten it further.

awenelo commented on lang/lang_en_fr.txt:
```diff
@@ -606,15 +598,15 @@
+#MSG_MMU_LOAD_FAILED c=20
 "MMU load failed     "
 "Echec chargement MMU"
 
```
```
Def. charg. MMU
```
(15)
"Fails" can be replaced by `Defis` ("Challenges") or `Défaut` ("Fault"), both of which can then be shortened to `Def.`

awenelo commented on lang/lang_en_fr.txt:
```diff
@@ -626,87 +618,87 @@
-#
+#MSG_MMU_POWER_FAILS c=15
 "MMU power fails"
 "Echecs alim. MMU"
```
```suggestion
"Defis alim. MMU"
```
(15)
Same as above, "fails" can be replaced by "challenges", however this time it doesn't need to be shortened.

carlin57 replied:
defis ?   def.  = défaut

awenelo replied:
Oh, I took `def` to mean `defis`. If we want to use `défaut`, we should probably find a different way to shorten it.

carlin57 replied:
it is the simplest by deduction (in France in the context of 3D printing) def. means "défaut " you can also use " faute " or " faux " but it sounds bizzard.

awenelo replied:
Ok, I guess that could work. I realized doesn't matter whether it's interpreted as `defis` or `défaut`, as both make sense in this context, so we'll use the shortened version for both.

```suggestion
"Def. alim. MMU"
```

Comment for line `617`/`609` has also been updated to match this discussion.

awenelo commented on lang/lang_en_fr.txt:
```diff
@@ -626,87 +618,87 @@
-#MSG_HIGH_POWER
+#MSG_HIGH_POWER c=10
 "High power"
 "Haute puiss"
```
```suggestion
"Haut.puiss"
```
(10)
`Haute` can be shortened to `Haut.`, by removing the space after `Haut.` we can get the length down to 10 without loss in meaning.

awenelo commented on lang/lang_en_fr.txt:
```diff
@@ -930,7 +922,7 @@
+#MSG_CALIBRATE_BED_RESET c=18
 "Reset XYZ calibr."
 "Reinit. calib. XYZ"
 
```
```
Reprend.impression
```
(18)
Like above, we can shorten `Reprendre` to `Reprend.`, and by removing the space, get it down to 18.

carlin57 replied:
"reprise" it's more appropriate. It's like taking it back or continuing.  "reprise" is not a conjugation but a feminine name is said "une reprise" ( reprendre)

awenelo replied:
In this context, I read `Reprise` as "take back", as in stop the print, more than "continue".

carlin57 replied:
sorry google translate

awenelo replied:
Oh, I see your edit to your previous comment. As we don't have enough space to indicate that it's the noun form of `reprise` we're using, not the verb form, `reprise` probably isn't the best choice. `Reprend.` still makes sense, and I think is clearer.

carlin57 replied:
just for example (Tomorrow I have to go back to work.):
"demain je dois reprendre le travail" or "demain je reprends le travail" = "demain reprise du travail" or "demain c'est la reprise du travail"

3d-gussner replied:
Which one should I use?

awenelo replied:
```
Reprise impression (🚀 )
```
I guess that we can probably use `reprise`, although a bit more difficult to understand (as it can be interpreted as "Take back the print"), context should help (being around "Stop print"). Although, there's not really a way to shorten this further, if ever necessary. (Would it be worthwhile to take a look into being able to scroll menu items?)

awenelo commented on lang/lang_en_fr.txt:
```diff
@@ -1022,71 +1014,67 @@
-#MSG_SOUND_ONCE
+#MSG_SOUND_ONCE c=7
 "Once"
 "Une fois"
```
```suggestion
"1 fois"
```
(6)
We can replace `Une` with the numeral `1` to get down to 6 characters, however this is less correct than using `Une`.

awenelo commented on lang/lang_en_fr.txt:
```diff
@@ -1098,39 +1086,39 @@
-#
+#MSG_SELECT_FILAMENT c=20
 "Select filament:"
 "Selectionnez le filament:"
```
```suggestion
"Choisi. le filament:"
```
(20)
Replacing `Selectionnez`(Select) with `Choisir`(Choose), then conjugating `Choisir` to `Choisissez`, which is then shortened to `Choisi.` gets us down to 20 characters. "Choose" sounds more like an instruction to choose a physical spool of filament, however, I think that the list of filaments below the instruction should make it clear enough.

carlin57 replied:
" Choix du filament"

awenelo commented on lang/lang_en_fr.txt:
```diff
@@ -946,51 +938,51 @@
-#MSG_TOSHIBA_FLASH_AIR_COMPATIBILITY
+#MSG_TOSHIBA_FLASH_AIR_COMPATIBILITY c=8
 "FlashAir"
 "FlshAir"
```
```suggestion
"FlashAir"
```
Is there a reason that this is translated to `FlshAir`? All the documentation I can find on FlashAir in french calls the card `FlashAir`.

3d-gussner replied:
Just changed it to use the origin message

awenelo commented on lang/lang_en_fr.txt:
```diff
@@ -946,51 +938,51 @@
-#MSG_FIND_BED_OFFSET_AND_SKEW_LINE1 c=60
+#MSG_FIND_BED_OFFSET_AND_SKEW_LINE1 c=20 r=2
 "Searching bed calibration point"
 "Recherche du point de calibration du plateau chauffant"
```
```suggestion
"Recherche point calibration du plateau"
```
In other places, we call the build plate `plateau`, instead of `plateau chauffant`, we can do the same here. Additionally, we can remove the first `du ` and `de ` without completely loosing the meaning of the sentence, however it now sounds more like a command to look up calibration points for the bed.

3d-gussner replied:
Still and issue because of text warp, see
```
[E]: Text is longer than definition on line 965: cols=20 rows=2 (rows diff=1)
 source text:
   1 |Searching bed       |
   2 |calibration point   |
 translated text:
   1 |Recherche point     |
   2 |calibration du      |
   3 |plateau             |
```

3d-gussner replied:
Is `Recherche point ca- libration du plateau` allowed? It would look like this
```
  -01234567890123456789-
1 |Recherche point ca- |
2 |libration du plateau|

awenelo replied:
As `ça` is a french word and with the dash, especially with the new line after it, I would read that something like "Research point to this: liberation(with a spelling mistake) of the plane(as in flat expanse, not plane, flying transportation)" 

Maybe we can use `Je cherche point du calibrage` ("Searching for calibration point") or `Je cherche point du calibrage plateau` ("Searching for the calibration point for the bed"), although neither of those option are as clear as `Recherche du point de calibration du plateau chauffant`.

awenelo reviewed (changes requested):
Most of the too-long translations are already hard to understand due to shortening that already exists, I've done my best to keep them legible, although some are bordering the edge of incomprehensibility.

I see that @carlin57 has also sent changes, I think that all the ones they sent are the same as mine, except for shortening `Defis` in `MSG_MMU_POWER_FAILS`, as the shortening isn't necessary and replacing `Reprendre`(Shortened to `Reprend.` by me) with `Repriser`(Conjugated to `Reprise`). To me, `Reprise` sounds more like re-do, not resume in this context.

Appologies about the double review, forgot to click "Add comment to review" when suggesting changes.

wavexx committed:
lang-check: respect --no-warning

wavexx committed:
lang-check: add basic punctuation checks

wavexx committed:
lang-check: attempt at spotting short translations

wavexx committed:
lang-check: cleanup

wavexx committed:
lang-check: add checks for % escapes

wavexx committed:
Two Italian translation improvements

wavexx committed:
lang-check: improve error strings

wavexx committed:
lang-check: handle/warn about trailing whitespace

wavexx committed:
Italian translation: remove trailing whitespace

3d-gussner committed:
Update German translation

3d-gussner committed:
Set MSG_SET_TEMPERATURE chars to c=20
Fix Italian warning shown by `lang-check.py`

3d-gussner commented:
@carlin57 @awenelo You guys are awesome :medal_sports: 
I will review these and come back to you.

Meanwhile the `lang-check.py` script has been polished and updated and here the results for French:
```
Start fr lang-check
[W]: Differing first punctuation character ([ => O) on line 29:
[W]: Differing last punctuation character (t => ]) on line 29:
 source text:
   |[0;0] point offset  |
 translated text:
   |Offset point [0;0]  |

[E]: Text is longer than definition on line 109: cols=18 rows=1 (rows diff=1)
 source text:
   |Bed level correct |
 translated text:
   |Corr. niveau plate|au

[E]: Text is longer than definition on line 601: cols=13 rows=1 (rows diff=1)
 source text:
   |Measured skew|
 translated text:
   |Deviat.mesure|e

[E]: Text is longer than definition on line 613: cols=15 rows=1 (rows diff=1)
 source text:
   |MMU load fails |
 translated text:
   |Echecs charg. M|MU

[E]: Text is longer than definition on line 641: cols=15 rows=1 (rows diff=1)
 source text:
   |MMU power fails|
 translated text:
   |Echecs alim. MM|U

[E]: Text is longer than definition on line 653: cols=10 rows=1 (rows diff=1)
 source text:
   |High power|
 translated text:
   |Haute puis|s

[E]: Text is longer than definition on line 933: cols=18 rows=1 (rows diff=1)
 source text:
   |Resume print      |
 translated text:
   |Reprendre impressi|on

[E]: Text is longer than definition on line 965: cols=20 rows=2 (rows diff=1)
 source text:
   1 |Searching bed       |
   2 |calibration point   |
 translated text:
   1 |Recherche du point  |
   2 |de calibration du   |
   3 |plateau chauffant   |

[E]: Text is longer than definition on line 1073: cols=7 rows=1 (rows diff=1)
 source text:
   |Once   |
 translated text:
   |Une foi|s

[E]: Text is longer than definition on line 1109: cols=20 rows=1 (rows diff=1)
 source text:
   |Select filament:    |
 translated text:
   |Selectionnez le fila|ment:

End fr lang-check
```

wavexx commented:
@3d-gussner At least for MSG_FIND_BED_OFFSET_AND_SKEW_LINE1 ("Searching bed calibration point"), it looks like the code could support up to 3 rows of text, but I didn't really test if that's the case.

In several of these cases it might be worth checking if the definition really needs to be that short and/or if it's missing

3d-gussner committed:
Update lang/lang_en_fr.txt

Co-authored-by: awenelo <56133692+awenelo@users.noreply.github.com>

3d-gussner committed:
Update French translation for MSG_BED_CORRECTION_MENU by @carlin57

3d-gussner committed:
Update lang/lang_en_fr.txt

Co-authored-by: awenelo <56133692+awenelo@users.noreply.github.com>

3d-gussner committed:
Update lang/lang_en_fr.txt

Co-authored-by: awenelo <56133692+awenelo@users.noreply.github.com>

3d-gussner committed:
Update lang/lang_en_fr.txt

Co-authored-by: awenelo <56133692+awenelo@users.noreply.github.com>

3d-gussner committed:
Update lang/lang_en_fr.txt

Co-authored-by: awenelo <56133692+awenelo@users.noreply.github.com>

3d-gussner committed:
Merge branch 'PFW-1189' of https://github.com/3d-gussner/Prusa-Firmware into PFW-1189

3d-gussner committed:
Update French translation for MSG_SELECT_FILAMENT as @carlin57 suggested

3d-gussner committed:
Use origin message for MSG_TOSHIBA_FLASH_AIR_COMPATIBILITY also in French

3d-gussner committed:
Updated French translation of MSG_MMU_LOAD_FAILED c=20 as @awenelo suggested

3d-gussner committed:
Update lang/lang_en_fr.txt

Co-authored-by: awenelo <56133692+awenelo@users.noreply.github.com>

3d-gussner committed:
Update lang/lang_en_fr.txt

Co-authored-by: awenelo <56133692+awenelo@users.noreply.github.com>

3d-gussner committed:
Merge branch 'PFW-1189' of https://github.com/3d-gussner/Prusa-Firmware into PFW-1189

3d-gussner committed:
Fix French translation for MSG_MMU_LOAD_FAILS

3d-gussner commented:
> @3d-gussner At least for MSG_FIND_BED_OFFSET_AND_SKEW_LINE1 ("Searching bed calibration point"), it looks like the code could support up to 3 rows of text, but I didn't really test if that's the case.
> 
> In several of these cases it might be worth checking if the definition really needs to be that short and/or if it's missing

Checked the source code and at first glance it looks like it should be no issue until i found this
```
		if (iteration > 0) {
			lcd_puts_at_P(0, next_line + 1, _i("Iteration "));////MSG_FIND_BED_OFFSET_AND_SKEW_ITERATION c=20
			lcd_print(int(iteration + 1));
```
Which would use lcd line 4 which doesn't exists and so create ugly glitches.

Not sure when this comes into place but I don't want touch it. @espr14 Do you have an idea how we could use 
- 3 lines for the text
- 1 line for the `x to 4` points and the `Iteration y` 
?

wavexx commented:
Maybe instead of adding lines, we can just overwrite the last line with "Iteration 2" and get those precious 3 lines

3d-gussner committed:
Use same wording in Spanish translation for "knob"

3d-gussner committed:
Fix typo in Spanish

3d-gussner commented:
> Maybe instead of adding lines, we can just overwrite the last line with "Iteration 2" and get those precious 3 lines

I never saw the "Iteration xx" so cant tell if it would fit after or could overwrite the `x of 4`

3d-gussner committed:
Remove white spaces in messags which saves ~62bytes

3d-gussner committed:
Revert that has been commited by mistake

Commit:
Fix some trailing whitespace and puctuation CZ

3d-gussner committed:
Remove white spaces in cz,nl,pl translations

3d-gussner committed:
Merge branch 'PFW-1189' of https://github.com/3d-gussner/Prusa-Firmware into PFW-1189

3d-gussner committed:
Remove white spaces and shortn some messages (inspired by FR, IT)

3d-gussner committed:
Add character ruler

espr14 commented:
> Not sure when this comes into place but I don't want touch it. @espr14 Do you have an idea how we could use
> 
> * 3 lines for the text
> * 1 line for the `x to 4` points and the `Iteration y`
>   ?
You can add the iteration text and number behind the " of 4" text if the iteration > 0. So it will not be printed on the 5th line. Moreover there could be fixed text "Iter. 2 of 2" because there are 2 iterations at most (now).

3d-gussner committed:
Update MSG_SEVER_SKEW and MSG_SLIGHT_SKEW message to c=14
Changed and tested Support-> "XYZ cal. details" LCD output

3d-gussner committed:
Fix character ruler for too many rows

wavexx committed:
lang-check: fancy ruler

3d-gussner committed:
Define MSG_FS_V_04_OR_NEWER and MSG_FS_V_03_OR_OLDER c=14

3d-gussner committed:
Define MSG_SELFTEST_FS_LEVEL

3d-gussner committed:
Update French translation MSG_RESUME_PRINT

3d-gussner committed:
Update `x of 4` and `x of 9` messages so longer text can be used.

3d-gussner commented:
@espr14 Can you please review https://github.com/prusa3d/Prusa-Firmware/pull/3079/commits/b8ae08fd3dd746a6f1ceb664393df0bec06d1468 ? This allows some longer translations.

3d-gussner committed:
Update #MSG_FIND_BED_OFFSET_AND_SKEW_LINE1 c=20 r=3

Commit:
Fix PL

Commit:
Fix FR

3d-gussner committed:
Merge branch 'PFW-1189' of https://github.com/3d-gussner/Prusa-Firmware into PFW-1189

3d-gussner commented:
> `Iter.` - with dot? Won't be translated?

Longest translation of `of` is at this moment 3 chars in German `von` and Dutch `van` and as you mentioned that it should be `Iter. 1 of 2` the translation `Iter` with a dot `Iter.` would exceed the 20 chars:
`1 von 4 Iter 1 von 2` is 20 chars long and fits in the 4th line.

Also I never saw the `Iteration` on the LCD and thought it wouldn't make sense to translate the already shorten English term `Iter`.
Should I change the `Iter x` to `Iteration x` or to `Iter x of 2`?

espr14 commented:
> Also I never saw the `Iteration` on the LCD and thought it wouldn't make sense to translate the already shorten English term `Iter`.

It happens very rarely that SuperPINDA is somehow biased at the start but it disappears over time. This causes bad data. So I added detection algorithm that triggers second iteration of calibration points' probing. Therefore it is not shown for the 1st iteration.

> Should I change the `Iter x` to `Iteration x` or to `Iter x of 2`?

You could also make it shorter/simpler: 'Iter. 2/2'.

3d-gussner commented:
> > Also I never saw the `Iteration` on the LCD and thought it wouldn't make sense to translate the already shorten English term `Iter`.
> > It happens very rarely that SuperPINDA is somehow biased at the start but it disappears over time. This causes bad data. So I added detection algorithm that triggers second iteration of calibration points' probing. Therefore it is not shown for the 1st iteration.
> 
> > Should I change the `Iter x` to `Iteration x` or to `Iter x of 2`?
> > You could also make it shorter/simpler: 'Iter. 2/2'.

Or remove the translation `of` completely  and just use `x/4 Iteration y/2`

espr14 commented:
Currently, there will be only the 2nd iteration of 2 visible. 1/1 will be hidden and there are 2 iterations at most (currently).

3d-gussner committed:
Revert "Fix FR"

This reverts commit a2fa7d34c57d9ae7ff69d0940be876d74d90a224.

Fixed with commits
https://github.com/prusa3d/Prusa-Firmware/pull/3079/commits/b8ae08fd3dd746a6f1ceb664393df0bec06d1468
https://github.com/prusa3d/Prusa-Firmware/pull/3079/commits/738a24178ac707a3b2122a88767e48a31be3ee46

3d-gussner committed:
Changed `x of 4` and `y of 9` to `x/4` and `y/4`
Added `Iteration` translation

3d-gussner committed:
Remove web links and `Prusa i3 MKxy OK.` from translations
as these never have been translated or doesn't make sense
to translate web links

3d-gussner commented:
@espr14 Here my latest version https://github.com/prusa3d/Prusa-Firmware/pull/3079/commits/686054b95e0ffdee7c547fff8f2daa9c2e610ec7 I have tested it and it looks good for me.

Commit:
Fix ES

DRracer commented on Firmware/Marlin_main.cpp:
```diff
@@ -3428,15 +3428,11 @@ bool gcode_M45(bool onlyZ, int8_t verbosity_level)
-			lcd_set_cursor(0, 2);
-			lcd_print(1);
-			lcd_puts_P(_T(MSG_FIND_BED_OFFSET_AND_SKEW_LINE2));
+			lcd_puts_at_P(0,3,_n("1/4"));
```
hmmm, extracting this piece of code into a separate variable makes the resulting code larger ... it looks like the compiler merged the "1/4" strings correctly

DRracer commented on Firmware/messages.c:
```diff
@@ -8,150 +8,150 @@
+const char MSG_BED[] PROGMEM_I1 = ISTR("Bed"); ////c=13
+const char MSG_BED_DONE[] PROGMEM_I1 = ISTR("Bed done"); ////c=20
+const char MSG_BED_HEATING[] PROGMEM_I1 = ISTR("Bed Heating"); ////c=20
+const char MSG_BED_LEVELING_FAILED_POINT_LOW[] PROGMEM_I1 = ISTR("Bed leveling failed. Sensor didn't trigger. Debris on nozzle? Waiting for reset."); ////c=20 r=6
```
have you tested this change? Does it change message formatting?

3d-gussner replied:
> Please verify if all reported \x00 texts are valid, some of them really look like untranslated for some reason (e.g. "Printer IP Addr" in CZ)

There are some and also found few others that have been missing. Question is if we should add these?

DRracer commented on Firmware/ultralcd.cpp:
```diff
@@ -2209,34 +2209,35 @@ void mFilamentItem(uint16_t nTemp, uint16_t nTempBed)
                 // handled earlier
                 break;
             }
+            lcd_puts_at_P(0, 3, _T(MSG_CANCEL)); ////MSG_CANCEL
```
why was this line added?

3d-gussner replied:
copy/paste error fixed in https://github.com/prusa3d/Prusa-Firmware/pull/3079/commits/30b383195dbcd2bfea94c074a746e0b99005ab38

DRracer commented on Firmware/ultralcd.cpp:
```diff
@@ -4624,8 +4625,8 @@ void lcd_v2_calibration()
-            _i("Select filament:"), ////c=20 r=1
-            _T(MSG_FILAMENT),_i("Cancel"));  ////c=19 r=1
+            _i("Select filament:"), ////MSG_SELECT_FILAMENT c=20
+            _T(MSG_FILAMENT),(_T(MSG_CANCEL)+1)); //Hack to reuse MSG but strip 1st char off
```
:+1:

DRracer reviewed (commented):
Please verify if all reported \x00 texts are valid, some of them really look like untranslated for some reason (e.g. "Printer IP Addr" in CZ)

3d-gussner committed:
Remove copy/paste error

3d-gussner committed:
Merge branch 'PFW-1189' of https://github.com/3d-gussner/Prusa-Firmware into PFW-1189

3d-gussner committed:
Remove PINDA from translations

3d-gussner committed:
Remove FINDA from translations

3d-gussner committed:
Remove FlashAir from translations

3d-gussner committed:
Remove "Firmware" from translations
deleted another unused "Firmware" define in language.h

3d-gussner committed:
Update few missing German translations
Fix Italian "SpoolJoin" to `\x00`

3d-gussner committed:
More missing German translations

3d-gussner committed:
More missing German translations

3d-gussner committed:
Update missing Dutch translations

vintagepc commented on lang/lang_en_nl.txt:
```diff
@@ -176,7 +176,7 @@
 #MSG_CHECKING_FILE c=17
 "Checking file"
-"\x00"
+"Control. bestand"
```
Does this really need abbreviation? seems like "Bestand controle" should be fine and still fit

3d-gussner committed:
Thanks to @vintagepc for reviewing Dutch translation

3d-gussner commented:
> Please verify if all reported \x00 texts are valid, some of them really look like untranslated for some reason (e.g. "Printer IP Addr" in CZ)

There are some and also found few others that have been missing. Question is if we should add these?

3d-gussner committed:
Update Dutch translation for `Cont.` as `Door.`

3d-gussner committed:
Remove `SpoolJoin` from translations

Commit:
Add missing CZ translations

Commit:
Merge branch 'PFW-1189' of github.com:3d-gussner/Prusa-Firmware into PFW-1189

Commit:
CZ Printer IP addr translation

Commit:
Add 2 missing translations for PL

still missing "Community made"

Commit:
PL 2

Commit:
IT temperatura

3d-gussner committed:
Updated few missing French Spanish translations

3d-gussner committed:
Merge branch 'PFW-1189' of https://github.com/3d-gussner/Prusa-Firmware into PFW-1189

3d-gussner committed:
Another missing Spanish translation

3d-gussner committed:
Update some French and Spanish missing translations

Commit:
PL "Community made"

wavexx committed:
More missing Italian translations

wavexx committed:
lang-check: use color names consistently

wavexx committed:
lang-check: Fix comment indentation

wavexx committed:
lang-check: add missing rulers

wavexx committed:
lang-check: optionally check for missing translations

Commit:
ES missing translations

Commit:
Looks like the last CZ message

3d-gussner committed:
Update all po files
